cloud/services/compute/network.go

```diff
@@ -196,3 +214,66 @@ func getRouterName(network string) string {
 func getRouterNatName(network string) string {
 	return fmt.Sprintf("%s-%s", network, "nat")
 }
+
+func (s *Service) getSubnetworkSpec(subnet *infrav1.SubnetSpec) *compute.Subnetwork {
+	res := &compute.Subnetwork{
+		EnableFlowLogs:        *subnet.EnableFlowLogs,
```

Comment:
is any of these fields are optional? what if they are not set from infrav1.subnetspec.

Reply:
No. All are required fields

Reply:
I mean from infrav1.SubnetSpec, some are optional, what if they are not provieded in infrav1.SubnetSpec

Reply:
Yes. infrav1.SubnetSpec does show it optional. But subnet creation fails if any of these is not provided: EnableFlowLogs, Region, NetworkUrl and cidrBlock. All of these are required fields. I didn't try without giving name so I am not sure about it.

Reply:
ok. I think some validation webhook should be added for this, later.